### ISSUE TITLE
References in block arrays

### ISSUE THREAD

**Issue by miracle2k** [sanity-io/gatsby-source-sanity#6]
Just played with the new version (v1), which is very nice, but the trick I used [here](https://github.com/sanity-io/gatsby-source-sanity/issues/5#issuecomment-449990546) doesn't work anymore. I am opening a new ticket, since it is strictly speaking a separate issue.

It is this: If my block has a reference to another object, it isn't included in `fooRaw`. Which I recognize is a problem - who knows how many levels deep I might want to follow. Using the old system, I could use a syntax trick in the way I define my own types in the Gatsby plugin to get the resolved reference.  

I wonder what can be done here?

Use arguments?

```
textRaw(followRefs: {depth: 3})
```

Allow me to query directly? As long as all the fields are always  available (e.g. `markDef`) could this possibly work? Using a fragment to query all the fields of a block necessary for render would be quite usuable, DX-wise. 

This has the added advantage of allowing me to not query certain fields of my own types that are included in a block content array: Fields that might contain private information, for example, which I do not want to be exposed. Thinking about how Gatsby works - AFAIK it exports and loads the GraphQL Query result for each page on navigation, thus making it public in full - hiding private information is not something that can be done with `textRaw`.

I also notice something else interesting going on. Given the following schema:

```
{
      title: 'Text',
      name: 'text',
      type: 'array',
      of: [
        {
          type: 'block',
          of: []
        },
        {type: 'quote'},
        {type: 'galleryWithText'},
        {type: 'reference', to: {type: 'keyPoint'}},
      ]
}
```

As long as I only use the `quote` and `galleryWithText` types, a query for 

```
children {
  id,
  __typename
},
text {
  _key
  _type            
}
```

has this response:

```
{
             children": [],
            "text": [
              {
                "_key": "c0eea66c0a54",
                "_type": "block"
              },
              {
                "_key": "12c6882e9317",
                "_type": "galleryWithText"
              }
            ]
}
```

But as soon as I add the reference type in the studio, the result becomes:

```
{
            "children": [
              {
                "id": "...",
                "__typename": "SanityBlock"
              },
              {
                "id": "...",
                "__typename": "SanityGalleryWithText"
              }
            ],
            "text": null
          }
```

For now I am going to remove the reference from the block array. It is not that important to me, and the UI in the studio isn't perfect either, but I do think it is worth supporting.

**Comment by kmelve** [comment]
Thanks for the feedback @miracle2k!

We're figuring out a good approach references in Portable Text with Gatsby. Stay tuned!

**Comment by Manicbsm27** [comment]
How to correct program in this topic....in this wrong format

**Comment by rexxars** [comment]
I've just released v1.1.0 with a new option for raw fields:
```graphql
{
  allSanityProject {
    edges {
      node {
        _rawTasks(resolveReferences: {maxDepth: 5})
      }
    }
  }
}
```
